runtime/criusupport/criusupport.cpp

```diff
@@ -341,6 +343,34 @@ releaseSafeOrExcusiveVMAccess(J9VMThread *currentThread, J9InternalVMFunctions *
 	}
 }
 
+static bool
+loadRestoreArguements(J9VMThread *currentThread, const char *optionsFile)
+{
+	bool result = true;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9JavaVMArgInfoList vmArgumentsList;
+	UDATA ignored;
+
+	vmArgumentsList.pool = pool_new(sizeof(J9JavaVMArgInfo), 0, 0, 0, J9_GET_CALLSITE(), J9MEM_CATEGORY_CLASSES, POOL_FOR_PORT(vm->portLibrary));
+	if (NULL == vmArgumentsList.pool) {
+		result = false;
+		goto done;
+	}
+	vmArgumentsList.head = NULL;
+	vmArgumentsList.tail = NULL;
+
+	if (0 != addXOptionsFile(vm->portLibrary, optionsFile, &vmArgumentsList, 0)) {
+		result = false;
+		goto done;
+	}
+
+	vm->checkpointState.restoreArgsList = createJvmInitArgs(vm->portLibrary, vm->vmArgsArray->actualVMArgs, &vmArgumentsList, &ignored);
```

Comment:
There is no issue at the default mode `-XX:+CRIURestoreNonPortableMode` here though multiple checkpoints could be prepared w/ a bit of cleanup before creating a new one.
```
		if (NULL != vm->checkpointState.restoreArgsList) {
			destroyJvmInitArgs(vm->portLibrary, vm->checkpointState.restoreArgsList);
		}
```